concrete_biopython/BioCircuit.py

```diff
@@ -0,0 +1,230 @@
+                 len_seqs,
+                 seq_interface,
+                 configuration,
+                 inputset=None,              
+                 seq_output = False,
+                 show_timing=True,
+                 **kwargs
+                ):
+        """
+        Arguments for the constructor:
+        
+        - `function` a function taking FheMutableSeq objects
+        - `len_seqs` a list of lengths of the Seq inputs that can be used by the circuit
+        - `seq_interface` a SeqInterface object to use for creating FheSeq objects
+        - `configuration` the configuration for the compiler
+        - `inputset` an optional custom inputset
+        - `seq_output` wether outputs are to be converted to Seq objects
+        - `show_timing` wether to show timing or not
+        - `**kwargs` any other named arguments for the compiler
+
+        TODO : - `encryption` a dictionnary with string keys and values equal to
+          "encrypted" or "clear" as required to compile a circuit
+        """
+
+        self._len_seqs = len_seqs
+        self._show_timing = show_timing        
+        self._seq_output = seq_output        
+        self._seq_interface = seq_interface        
+
+        # create an encryption dictionnary from the number of input sequences, where everything is encrypted
+        encryption = { "seq" + str(i): "encrypted" for i in range(1,len(len_seqs)+1) }
+
+        # Create the circuit inputset from the seq lengths and the maximum integer
+        # Use seq_interface.max_integer() to know the maximum integer that can be
+        # used to represent a character in FheSeq obects        
+        if inputset is not None:
+            # check correctness of inputset
+            if not ( isinstance(inputset, list)
+                and all(isinstance(item, tuple) and all(isinstance(seq_item, (Seq, MutableSeq))
+                    for seq_item in item) for item in inputset)):
+                raise ValueError("The inputset should be either None or with type List[Tuple[Union[Seq, MutableSeq]]]")
+
+            inputset = [tuple([seq_interface.to_integers(seq_item) for seq_item in item]) for item in inputset]
+
+        else:
+            inputset = [tuple([np.random.randint(0, seq_interface.max_integer()+1, size=(len_seq,))
+                                for len_seq in len_seqs]) for _ in range (300)]
+
+        # Wrap the function so that it can use any number of MutableSeq objects in inputs and a MutableSeq/Seq in output
+        # get parameter names from encryption directory        
+        param_names = list(encryption.keys())
+        wrapped_function = function_double_wrapper_factory(param_names)(function, seq_interface)
+
+        # compile the circuit   
+        compiler = fhe.Compiler(wrapped_function, encryption)
+
+        compile_ = lambda _ : compiler.compile (
+                inputset = inputset,
+                configuration = configuration,
+                **kwargs,
+            )
```

Comment:
What's the purpose of empty parameter, you can do:
```python
lambda: compiler.compile (inputset = inputset, configuration = configuration, **kwargs)
```
and call without any arguments.

Reply:
that's because measure_time cannot take no arguments

Reply:
It has this signature:
```python
def measure_time(function, descripton, *inputs):
    ...
```
and it can take no input arguments.

You can call it like so:
```python
measure_time(function_without_args, "Foo ")
```

Reply:
you are using a trick as well, mine works fine :)

Reply:
yes okay thanks :)